### PR TITLE
Add --legacy40 flag for evolven3fit_new to use the grid for nf=3 instead of nf=4

### PR DESCRIPTION
**Pull request by scarlehoff**
So that the `eko` that are being used to produce many of the results can easily be reproduced with `evolven3fit_new`.

This deals with #1856.

### CODE REVIEW

File: n3fit/src/evolven3fit_new/utils.py [hunk 1]
```diff
@@ -178,15 +178,17 @@ def get_theoryID_from_runcard(usr_path):
...
+def generate_q2grid(Q0, Qfin, Q_points, match_dict, nf0=None, ic=False):
     """Generate the q2grid used in the final evolved pdfs or use the default grid if Qfin or Q_points is
     not provided.
 
     match_dict contains the couples (mass : factor) where factor is the number to be multiplied to mass
     in order to obtain the relative matching scale.
     """
     if Qfin is None and Q_points is None:
-        if nf0 == 4:
+        if nf0 == 4 and ic:
```
Comment (giacomomagni) [inline n3fit/src/evolven3fit_new/utils.py:189]:
```suggestion
        if nf0 == 3 or ic:
```
this way you can remove the last branch.

File: n3fit/src/n3fit/scripts/evolven3fit_new.py [hunk 2]
```diff
@@ -34,6 +34,9 @@ def construct_eko_parser(subparsers):
     parser.add_argument(
         "-p", "--x-grid-points", default=None, type=int, help="Number of points of the x-grid"
     )
+    parser.add_argument(
+        "--ic", action="store_true", help="Use the nf=3 grid even if the fit was run with nf=4"
```
Comment (giacomomagni) [inline n3fit/src/n3fit/scripts/evolven3fit_new.py:38]:
```suggestion
        "--ic", action="store_true", help="Use the nf0=3 grid even if the fit was run with nf0=4"
```

File: n3fit/src/evolven3fit_new/utils.py [hunk 3]
```diff
@@ -178,15 +178,17 @@ def get_theoryID_from_runcard(usr_path):
...
+            return Q2GRID_Nf03
+        elif nf0 == 4:
             return Q2GRID_Nf04
         elif nf0 == 3:
             return Q2GRID_Nf03
```
Comment (scarlehoff) [inline n3fit/src/evolven3fit_new/utils.py:194]:
```suggestion
```

**Review by giacomomagni — APPROVED** [review]

**Comment by scarlehoff** [comment]
Ok, given the discussion let me change this to:

1. Use always a "complete down to 1 GeV grid'
2. Add a few extra points before 1.65 GeV that are basically the same that are already in the nf=3

And remove the different conditions. This means the result of `evolven3fit` and `evolven3fit_new` would be different until I update the 4.0 theories but I'll do that little by little in the next few weeks (and theory 200 is already done so the most important one is safe).

**Review by giacomomagni — COMMENTED** [review]
LGTM. Stupid question: i'd have to redo also my ekos if I've used `Q2GRID_Nf03` before right?

**Comment by scarlehoff** [comment]
>  i'd have to redo also my ekos if I've used Q2GRID_Nf03 before right?

I'd say not urgent (and specially not for N3LO). 
The problem of having used nf=3 is that you would have some differences at the boundaries (because they happen at slightly different values of Q). But this kind of 1-to-1 comparison might be relevant (if it ever is) only for QED which is really supposed to be "4.0 with the photon".

**Comment by scarlehoff** [comment]
With `--legacy40` the resulting LHAPDF grid file agrees with the apfel evolution at 1e-3 with some bigger than that differences in the thresholds and around them (Q ~ 4.92 GeV). This differenced was already there in the past and @AleCandido and @felixhekhorn convinced me that the eko result was correct then so I'm happy to go with it.

The default eko should be finish soon, then I'll check that the results above 1.65 GeV are also consistent and this can be merged.

**Pull request merged**